sql/src/main/java/io/crate/operation/user/User.java

```diff
@@ -57,14 +58,28 @@ public String name() {
         return privileges;
     }
 
+    public boolean isSuperUser() {
+        return roles.contains(Role.SUPERUSER);
+    }
+
+    public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, Privilege.State state) {
+        Optional<Privilege> hasPrivilege = privileges.stream()
```

Comment:
@m-kharbat I think we should pass in here the complete Privilege which is required (Clazz.TABLE and tableName as Ident) and solve the hierarchy check here inside. lets discuss how we can realize that..